packages/hdwallet-web3-portis/src/adapter.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    Keyring,
+    HDWallet
+  } from '@shapeshiftoss/hdwallet-core'
+
+import { PortisTransport } from './PortisTransport'
+import { Web3PortisHDWallet } from './web3Portis'
+
+  export class Web3PortisAdapter {
+    keyring: Keyring
+  
+    private constructor (keyring: Keyring) {
+      console.log('portis adapter constructor')
+      this.keyring = keyring
+    }
+  
+    public static useKeyring (keyring: Keyring) {
+      console.log('portis adapter useKeyring')
+      return new Web3PortisAdapter(keyring)
+    }
+  
+    public async initialize (portis: any): Promise<number> {
+      console.log('portis adapter initialize')
+      let transport = new PortisTransport(this.keyring)
```

Comment:
If you're going to have a transport, you should put the `portis` object in the transport. Otherwise it's an unnecessary abstraction.